test/e2e/framework/util.go

```diff
@@ -4396,7 +4396,7 @@ func OpenWebSocketForURL(url *url.URL, config *restclient.Config, protocols []st
 	if err != nil {
 		return nil, fmt.Errorf("failed to create tls config: %v", err)
 	}
-	if tlsConfig != nil {
+	if tlsConfig != nil || url.Scheme == "https" {
```

Comment:
As you pointed on the issue report, tlsConfig: nil condition is like https://github.com/kubernetes/kubernetes/blob/5e86fa43f59f8ed4fd353ab17dc1e65708a9ae71/staging/src/k8s.io/client-go/transport/transport.go#L56
So doesn't your kubeconfig contain `certificate-authority-data: ...`, `insecure-skip-tls-verify: true` or the other related options?

Reply:
No, my kubeconfig just looks like this:

```yaml
apiVersion: v1
clusters:
- cluster:
    server: https://cluster-1.example.org
  name: cluster-1
contexts:
- context:
    cluster: cluster-1
    namespace: default
    user: <name>
  name: cluster-1
current-context: cluster-1
kind: Config
preferences: {}
users:
- name: <name>
  user:
    token: <token>
```

The server endpoint is an AWS ELB with a "real" certificate already trusted by most systems.

Reply:
> The above kubeconfig seems that clients(kubectl, e2e) don't verify the api server, right?

They do verify it because they use the trusted CAs from the OS (because of how golang net HTTP client works). Setting `insecure-skip-tls-verify: true` would reduce the security for no reason other than to work around the fact that the e2e tests currently makes too strong assumptions about my Kubernetes setup. i.e. that the apiserver is served with a self-signed certificate.

Reply:
Thanks for your explanation :-)
https://tools.ietf.org/html/rfc6455#section-3 says `The port component is OPTIONAL; the default for "ws" is port 80,
   while the default for "wss" is port 443.` So the schema "https" should be changed like this as line 4402.

Reply:
I don't quite follow. Do you want me to remove the code for setting the default port? I can do it, but should it be done in this PR? It's not completely related to the fix I did IMO.

Reply:
Sorry for mess comment. I just wanted to say this change seems good for me as the default port of wss. Then I did put `/lgtm`. It is fine to wait for another review for merging this without anything.

Reply:
Ok, thanks a lot for the review!